src/session_impl.cpp

```diff
@@ -697,15 +704,15 @@ namespace aux {
 			if (val) m_settings.set_int(settings_pack::allowed_enc_level, val.int_value());
 		}
 #endif
-
+		bool need_apply = false;
 		if (flags & session::save_settings)
 		{
-			settings = e->dict_find_dict("settings");
+			settings = e.dict_find_dict("settings");
 			if (settings)
 			{
+				load_pack_from_dict(settings, pack);
```

Comment:
are you relying in ``load_pack_from_dict`` to augment the pack? so it's an in-out parameter?